src/yafowil/common.py

```diff
@@ -308,13 +333,18 @@ def input_attributes_full(widget, data, value=None):
     return input_attrs
 
 
-@managedprops(*css_managed_props)
-def input_generic_renderer(widget, data, custom_attrs={}):
+@managedprops(*managed_props__input_full)
+def input_generic_renderer(widget, data, pos='before', custom_attrs={}):
     """Generic HTML ``input`` tag render.
     """
     input_attrs = input_attributes_full(widget, data)
     input_attrs.update(custom_attrs)
-    return data.tag('input', **input_attrs)
+    rendered = data.tag('input', **input_attrs)
+    if pos == 'before':
+        rendered = rendered + (data.rendered or u'')
+    else:
+        rendered = (data.rendered or u'') + rendered
+    return rendered
```

Comment:
``input_generic_renderer`` is supposed only to render an input tag, no matter whether data already contains rendered markup, please revert